docs/setup/sso.md

```diff
@@ -0,0 +1,298 @@
+
+This section contains sample configurations for popular OIDC providers.
+
+### Authentik
+
+[Authentik](https://goauthentik.io/) is an open-source IdP solution.
+
+1. Create a provider in Authentik, with type OAuth2/OpenID.
+2. The parameters are:
+  - Client Type: Confidential
+  - Redirect URIs: `https://<auth-service-domain>/upstream/callback/<id>`
+3. Create an application for the authentication service in Authentik and link it to the provider.
+4. Note the slug of your application, Client ID and Client Secret.
+
+Authentication service configuration:
+
+```yaml
+upstream_oauth2:
+  providers:
+    - id: 01HFRQFT5QFMJFGF01P7JAV2ME
+      human_name: Authentik
+      issuer: "https://<authentik-domain>/application/o/<app-slug>/" # TO BE FILLED
+      client_id: "<client-id>" # TO BE FILLED
+      client_secret: "<client-secret>" # TO BE FILLED
+      scope: "openid profile email"
+      claims_imports:
+        localpart:
+          action: require
+          template: "{{ user.preferred_username }}"
+        display_name:
+          action: suggest
+          template: "{{ user.name }}"
+        email:
+          action: suggest
+          template: "{{ user.email }}"
+          set_email_verification: always
+```
+
+
+### Facebook
+
+0. You will need a Facebook developer account. You can register for one [here](https://developers.facebook.com/async/registration/).
+1. On the [apps](https://developers.facebook.com/apps/) page of the developer console, "Create App", and choose "Allow people to log in with their Facebook account".
+2. Once the app is created, add "Facebook Login" and choose "Web". You don't
+   need to go through the whole form here.
+3. In the left-hand menu, open "Use cases" > "Authentication and account creation" > "Customize" > "Settings"
+   * Add `https://<auth-service-domain>/upstream/callback/<id>` as an OAuth Redirect URL.
+4. In the left-hand menu, open "App settings/Basic". Here you can copy the "App ID" and "App Secret" for use below.
+
+Authentication service configuration:
+
+```yaml
+upstream_oauth2:
+  providers:
+    - id: "01HFS3WM7KSWCEQVJTN0V9X1W6"
+      issuer: "https://www.facebook.com"
+      human_name: "Facebook"
+      brand_name: "facebook"
+      discovery_mode: disabled
+      pkce_method: always
+      authorization_endpoint: "https://facebook.com/v11.0/dialog/oauth/"
+      token_endpoint: "https://graph.facebook.com/v11.0/oauth/access_token"
+      jwks_uri: "https://www.facebook.com/.well-known/oauth/openid/jwks/"
+      token_endpoint_auth_method: "client_secret_post"
+      client_id: "<app-id>" # TO BE FILLED
+      client_secret: "<app-secret>" # TO BE FILLED
+      scope: "openid"
+      claims_imports:
+        localpart:
+          action: ignore
+        displayname:
+          action: suggest
+          template: "{{ user.name }}"
+        email:
+          action: suggest
+          template: "{{ user.email }}"
+          set_email_verification: always
+```
+
+
+## GitLab
```

Comment:
s/## GitLab/### GitLab/

Reply:
fyi if you hit the "Add a suggestion" button in the GitHub markdown editor, it formats the change like this:

```suggestion
### GitLab
```

which I can then one-click commit with you as co-author